databricks/koalas/indexes.py

```diff
@@ -130,7 +120,11 @@ def _with_new_scol(self, scol: spark.Column) -> "Index":
         :param scol: the new Spark Column
         :return: the copied Index
         """
-        return Index(self._kdf, scol=scol)
+        sdf = self._internal.spark_frame.select(scol)
+        internal = _InternalFrame(
+            spark_frame=sdf, index_map=OrderedDict(zip(sdf.columns, self._internal.index_names))
+        )
+        return DataFrame(internal).index
```

Comment:
Hm .. I think it will trigger the operations on different DataFrames among the operations in Index .. e.g.)

```python
import databricks.koalas as ks
index = ks.range(100).index
index + index + index
```

There looks another issue in `compute.ops_on_diff_frames` ...  operations against the indexes from different frames are not supported ..

Reply:
Good point.

Reply:
Good point.

The way we manage index Spark columns is different from Series, so we need further refactoring in `_InternalFrame`, e.g., we should manage them as Spark columns instead of Spark column names. The way in Series is not good enough yet, though.

Actually creating new DataFrame as an anchor is better than now compared to pandas behavior:

```py
>>> pdf = pd.DataFrame({'a':[1,2,3]})
>>> pidx = pdf.index
>>> pidx.name = 'i'
>>> pdf
   a
i
0  1
1  2
2  3
>>> (pidx + pidx).name = 'ii'
>>> pdf
   a
i
0  1
1  2
2  3
```

When operating `pidx + pidx`, the connection to the original `pdf` seems dropped.

Whereas:

```py
>>> kdf = ks.DataFrame({'a':[1,2,3]})
>>> kidx = kdf.index
>>> kidx.name = 'i'
>>> kdf
   a
i
0  1
1  2
2  3
>>> (kidx + kidx).name = 'ii'
>>> kdf
    a
ii
0   1
1   2
2   3
```

It updates index name even after `kidx + kidx`.